repair/row_level.cc

```diff
@@ -1576,7 +1636,10 @@ class repair_meta {
         rlogger.debug("<<< Finished Row Level Repair (Follower): local={}, peers={}, repair_meta_id={}, keyspace={}, cf={}, range={}",
             utils::fb_utilities::get_broadcast_address(), from, repair_meta_id, ks_name, cf_name, range);
         auto rm = get_repair_meta(from, repair_meta_id);
-        return remove_repair_meta(from, repair_meta_id, std::move(ks_name), std::move(cf_name), std::move(range));
+        rm->set_repair_state_for_local_node(repair_state::row_level_stop_started);
+        return remove_repair_meta(from, repair_meta_id, std::move(ks_name), std::move(cf_name), std::move(range)).then([rm] {
+            rm->set_repair_state_for_local_node(repair_state::row_level_stop_finished);
```

Comment:
Is the `rm` alive here still?

Reply:
I think it is alive. The call rm->set_repair_state_for_local_node does not yield.   We should be able to access rm inside the lambda: [rm] { rm->foo() }.

Reply:
Won't `remove_repair_meta()` destroy it? We are in a continuation attached to that here.

Reply:
```
 914     static future<>
 915     remove_repair_meta(const gms::inet_address& from,
 916             uint32_t repair_meta_id,
 917             sstring ks_name,
 918             sstring cf_name,
 919             dht::token_range range) {
 920         node_repair_meta_id id{from, repair_meta_id};
 921         auto it = repair_meta_map().find(id);
 922         if (it == repair_meta_map().end()) {
 923             rlogger.warn("remove_repair_meta: repair_meta_id {} for node {} does not exist", id.repair_meta_id, id.ip);
 924             return make_ready_future<>();
 925         } else {
 926             auto rm = it->second;
 927             repair_meta_map().erase(it);
 928             rlogger.debug("remove_repair_meta: Stop repair_meta_id {} for node {} started", id.repair_meta_id, id.ip);
 929             return rm->stop().then([rm, id] {
 930                 rlogger.debug("remove_repair_meta: Stop repair_meta_id {} for node {} finished", id.repair_meta_id, id.ip);
 931             });
 932         }
 933     }
```
It only removes the lw_shared_ptr from repair_meta_map(), we still have a reference to rm, so it will not be destroyed.

Reply:
I see.